Doc/library/stdtypes.rst

```diff
@@ -1645,9 +1645,10 @@ expression support in the :mod:`re` module).
 .. method:: str.endswith(suffix[, start[, end]])
 
    Return ``True`` if the string ends with the specified *suffix*, otherwise return
-   ``False``.  *suffix* can also be a tuple of suffixes to look for.  With optional
-   *start*, test beginning at that position.  With optional *end*, stop comparing
-   at that position.
+   ``False``.  *suffix* can also be a tuple of suffixes to look for.
+
+   Optional arguments *start* and *end* are interpreted as in slice notation,
+   and restrict the comparison to the range [*start*, *end*].
```

Comment:
Shouldn't it be a half-open range?
```suggestion
   and restrict the comparison to the range [*start*, *end*).
```

Personally I still find this a bit hard to understand. What if we added that `s.endswith(suffix, start, end)` is equivalent to `s[start:end].endswith(suffix)`?

Reply:
Maybe this is closer to the other ones:

.. method:: str.startswith(prefix[, start[, end]])

   Return ``True`` if the range [*start*, *end*] starts with the *prefix*, 
   otherwise return ``False``. *prefix* can also be a tuple of prefixes to 
   look for.  Optional arguments *start* and *end* are interpreted as in 
   slice notation.